### ISSUE TITLE
Raspberry pi: glitching 2D fighting games with threaded rendering

### ISSUE THREAD

**Issue by sergioad** [libretro/flycast#997]
Hello friends when playing SFIII on Flycast with the "Threaded rendering" option enabled I am getting lots of issues on my Raspberry pi 4, this issue apparently also aplies in other 2D fighting games like CvS 1, CvS 2 and MvC2

**Comment by barbudreadmon** [comment]
I can't reproduce this issue with the information you provided (i don't own a pi4 though), so i'm considering 2 possibilities : 
- this is related to another setting you changed, especially make sure you didn't set resolution to 320x240, downscaling those 2D games is known for causing gfx glitches
- that's yet another pi4 gpu driver bug, and not really our problem as far as i'm concerned, those rpi fundation guys are paid to write their gpu driver

**Comment by sergioad** [comment]
Hello @barbudreadmon thanks for the super quick response, I have not set anything different, and let me tell you... while disabling threaded rendering highly improves Street fighter III sadly it worsens Capcom vs SNK 2 as the music is chopped here and there

Regarding my settings everything was set to the defaults including the resolution

BTW how is that Vulkan support going? When the rpi Vulkan gets integrated to the next version of the video drivers I will surely play Flycast + vulkan + my 1950 MHz overclock

**Comment by barbudreadmon** [comment]
> everything was set to the defaults including the resolution

Could you confirm it's currently set to 640x480 ? i think some distro were changing that default setting, so maybe retropie too

**Comment by sergioad** [comment]
It is, I will check out the scaling options to see if the aspect ratio was charged like they did on PSP and GBA

**Comment by sergioad** [comment]
> > everything was set to the defaults including the resolution
> 
> Could you confirm it's currently set to 640x480 ? i think some distro were changing that default setting, so maybe retropie too

**Comment by sergioad** [comment]
> Could you confirm it's currently set to 640x480 ? i think some distro were changing that default setting, so maybe retropie too

I already tried using the core provided aspect ratio and the problem persists, it might start perfectly fine but the graphics start to glitch after executing lots of super moves
![Street Fighter III - 3rd Strike v1 001 (2000)(Capcom)(NTSC)(US) ! -201228-155700](https://user-images.githubusercontent.com/5375145/103245611-534be600-4926-11eb-8c10-04b2059f0d97.png)

**Comment by barbudreadmon** [comment]
I managed to reproduce this with your instructions (well, it didn't look as bad though), it seems to only happens with per-triangle alpha sorting, i couldn't reproduce with the other alpha sorting choices.

**Comment by sergioad** [comment]
In the pi you only have per triangle and other (I do not remember the name) which is less accurate

**Comment by barbudreadmon** [comment]
> which is less accurate

Yeah, and it doesn't seem to happen when using this one.

**Comment by sergioad** [comment]
No, I only get some screen tearing when I dash a lot using per strip alpha sorting

It is very noticeable in Guilty Gear X but also occurs in other games like CvS 1 & 2

**Comment by sergioad** [comment]
Very noticeable in Guilty Gear X

**Comment by sergioad** [comment]
Any idea how to solve it? Three months in and does not seem to be solved in updates, if I can help please let me know
